include/arch/riscv/arch/sbi.h

```diff
@@ -43,6 +54,25 @@
 #define SBI_REMOTE_SFENCE_VMA_ASID 7
 #define SBI_SHUTDOWN 8
 
+
+
+#ifdef CONFIG_SEL4_RV_MACHINE
+extern char trap_entry[];
+#define SBI_CALL(which, arg0, arg1, arg2) ({ 			\
+        unsigned long mtvec = read_csr(mtvec);                 \
+        write_csr(mtvec, 0x0000000080000004ul);                     \
```

Comment:
Is this a spike specific physical address?

Reply:
Spike + riscv-pk. It's the value riscv-pk sets before going loading the payload. Yes, it's physical. I made a change to save riscv-pk's mtval on boot time. This should make seL4/spike plat more portable in case this address has ever changed by pk/spike (for any reason).

https://github.com/riscv/riscv-pk/blob/master/machine/mentry.S#L34